docs/src/configuring-zed.md

```diff
@@ -1150,6 +1150,136 @@ To interpret all `.c` files as C++, files called `MyLockFile` as TOML and files
 }
 ```
 
+## Diagnostics
+
+- Description: Configuration for diagnostics-related features.
+- Setting: `diagnostics`
+- Default:
+
+```json
+{
+  "diagnostics": {
+    "include_warnings": true,
+    "inline": {
+      "enabled": false
+    }
+    "update_with_cursor": false,
```

Comment:
This is the best part of the impl to me, honestly, I wonder if we could 
* move it into `inline` as this settings knob is related to this part of the functionality only? (apart from the fact that it spoils f8 navigation sometimes 🙂, see another comment )
* enable it by default, and explain that when disabled, same effect could be achieved with f8/shift-f8 or the new action (if we decide to leave it)

Reply:
I'm absolutely on board enabling by default, but not at all on board moving it to the inline settings. I actually think this almost a separate feature because its so useful for the whole "what's that squiggly line" sort of interaction.

I'm suddenly wondering if I don't have a bug here. I very much had a few because it was originally tied to the inline diagnostics implementation but I thought I removed the dependency there. Regardless, I definitely agree that this particular behavior is probably something we should add regardless of the inline diagnostics aspect.